homeassistant/components/slack/notify.py

```diff
@@ -74,16 +82,55 @@
 )
 
 
-async def async_get_service(hass, config, discovery_info=None):
+class AuthDictT(TypedDict, total=False):
+    """Type for auth request data."""
+
+    auth: BasicAuth
+
+
+class FormDataT(TypedDict):
+    """Type for form data, file upload."""
+
+    channels: str
+    filename: str
+    initial_comment: str
+    title: str
+    token: str
+
+
+class MessageT(TypedDict, total=False):
+    """Type for message data."""
+
+    link_names: bool
+    text: str
+    username: str  # Optional
+    icon_url: str  # Optional
+    icon_emoji: str  # Optional
+    blocks: List[Any]  # Optional
+
+
+async def async_get_service(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+) -> Optional[SlackNotificationService]:
     """Set up the Slack notification service."""
-    session = aiohttp_client.async_get_clientsession(hass)
-    client = WebClient(token=config[CONF_API_KEY], run_async=True, session=session)
+    session: ClientSession = aiohttp_client.async_get_clientsession(hass)
+    client: WebClient = WebClient(
+        token=config[CONF_API_KEY], run_async=True, session=session
+    )
 
     try:
         await client.auth_test()
     except SlackApiError as err:
-        _LOGGER.error("Error while setting up integration: %s", err)
-        return
+        _LOGGER.error("Error while setting up integration: %r", err)
+        return None
+    except ClientError as err:
+        _LOGGER.warning(
+            "Error testing connection to slack: %r"
+            "Continuing setup anyway, but notify service might not work",
```

Comment:
Tell me the intention here – is the idea that if Slack fails "temporarily" during service setup, it may work down the road? Is there a scenario in which this occurs?

Reply:
I recently had no internet connection at home and noticed, after a restart, the error case. Since slack wasn't setup, it caused all automation to fail that at one point call the `notify` service, even if they also do some local stuff too.

Additionally I really wanted to add an automation to notify me once the connection was restored but that would have required to restart HomeAssistant first.

As a point of comparison, I took a look at the `mobile_app` integration. It just always registers the service. In cases where the service call is unsuccessful, it only prints an error and moves on. That way all automations continue working.

Reply:
Great! 👍🏻